src/Orleans.Runtime/GrainDirectory/DirectoryMembershipSnapshot.cs

```diff
@@ -62,6 +62,11 @@ private static int CompareSiloAddressesForRing(SiloAddress left, SiloAddress rig
         /// </summary>
         public ClusterMembershipSnapshot ClusterMembership { get; }
 
+        /// <summary>
+        /// The number of active silos.
+        /// </summary>
+        public int ActiveMemberCount { get; }
```

Comment:
I made the distinction here just because we may implement what we were discussing with each silo owning multiple points on the ring to improve balance. If we did that then `ring.Count` may no longer be consistent with what this property represents.

The info cannot get out of sync: this is an immutable class